### PR TITLE
openstack UPI: Automate Network creation

### PR DESCRIPTION
**Pull request by pierreprinetti**
Create an Ansible playbook for deploying the Networks required for a
User Provided Infrastructure installation.

Implements https://jira.coreos.com/browse/OSASINFRA-784

### CODE REVIEW

**Comment by pierreprinetti** [comment]
@mandre 
Like you suggested, I have added a `down` script for teardown. It has proven useful for development :grin:

File: upi/openstack/inventory.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+all:
+  hosts:
+    localhost:
+      ansible_connection: local
+      ansible_python_interpreter: "{{ansible_playbook_python}}"
+
+      os_subnet_range: '192.0.2.0/24'
```
Comment (tomassedovic) [inline upi/openstack/inventory.yaml]: This IP range is only supposed to be used for examples and documentation and the RFC encourages routers etc. to drop any traffic going through it. This is fine for the UPI doc but not as a default config for these playbooks.

I think we should either always require the caller to set this (no default -- and then we can mention the range above in the docs) or use one that should work (barring IP conflicts).

The default IPI MachineCIDR range is `10.0.0.0/16` -- I think we can just use the same here.

**Comment by tomassedovic** [comment]
/lgtm
/hold

Looking good! I've had an old version of `ansible` and `openstacksdk` which didn't work, but the new readme change made provides the necessary versions so that's good!

I'd also like to see some docs on how to run the playbooks and pass in the inventory and how to destroy the resources. It would also be nice keep the deps as a requirements.txt file checked in this repo (so we can update it and people can automate that). But none of those are necessary.

Feel free to `/hold cancel` once the `e2e-openstack` test passes.

**Comment by pierreprinetti** [comment]
@tomassedovic 
I should have addressed your comments. PTAL!

**Comment by tomassedovic** [comment]
/lgtm

Perfect, thanks! Feel free to remove the hold once the e2e-openstack job passes.

**Comment by pierreprinetti** [comment]
/hold cancel

`e2e-openstack` has succeeded.

**Pull request merged**